### ISSUE TITLE
installing and upgrading python3

### ISSUE THREAD

**Issue by svenevs** [Homebrew/brew#4151]
Hello,

I have had some very strange issues happen with python, but cannot determine what I possibly could have done to arrive here (or how you would replicate the issue).  There are two distinct problems, one is about upgrading, and one is about installing anew.

1. Autoupdating python is incomplete, and causes trauma.  I use brew python and pip as encouraged -- I `pip3` install most things, which will install to e.g. `/usr/local/lib/python-3.6`.  The problem is that without me really even realizing it, my python got upgraded by brew automatically from `3.6.4` to `3.6.5`.  Most of my local development was broken, and I believe the issue here is that things I installed with `pip3` when using `3.6.4` are not guaranteed to work with `3.6.5`.

    To be clear, I really don't know if that is true.  However, I simply deleted everything under `/usr/local/lib/python-3.6` and then reinstalled everything I wanted with `pip3` (from python `3.6.5`) and things worked again.

    If this is indeed a problem, I see two possible solutions that would be relatively easy to implement:

    - Do not automatically update `python` installations.  This is my preference.
        - I know I can tell brew [not to update globally](https://github.com/Homebrew/brew/issues/1670), is there a way for me to say blacklist just python (I like the auto update in general!).
    - Delete `/usr/local/lib/python3.6/`.  This sounds really dangerous to do automatically.

2. Since the `3.6.5` update (maybe a week or two ago for me), I've force uninstalled things and restarted twice now.  Everything works and life is good, and then one day out of the blue I cannot run the brew installed python 3.6.5, with "PyInitialize unable to load the file system codec".

    I do not understand why things work for a while, and then mysteriously (with or without reboots in between), the system codec issue appears.  A full reinstallation will solve it, and then it will break again.

    This is far beyond my understanding of Python, but I couldn't help but notice that the poured bottle was compiled with a different XCode toolchain than mine.  It would report `GCC 4.2.1 Compatible Apple LLVM 9.0.0` (or some other number, but `9.x` being the key here), whereas I deliberately have (and will need to keep) XCode 8.x toolchain for the foreseeable future.

    Would that somehow cause this system codec issue?  I just went through and `--build-from-source` installed, so Python now reports `GCC 4.2.1 Compatible Apple LLVM 8.0.0`.

I thought I would report the issue, but other than `brew doctor` reporting an outdated XCode, everything else is working great.  I've never encountered such strange problems in all the years i've been using brew, so can't help but wonder if it is something I did locally somehow.

In short, I don't know how you would reproduce this issue, but thought I would at least include my needed steps to fix it for myself, and identify two potential causes that I am unsure of.

Thank you for reading, if you feel the issue is inappropriate feel free to close without response.

**Comment by MikeMcQuaid** [comment]
From the issue template:

> **Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you do not read fill out the issue template repeatedly you will be blocked from ever submitting issues to Homebrew again.**

If you add the necessary information to this issue ((don't create a new one, please) then this issue may be reopened.

**Comment by svenevs** [comment]
Hi @MikeMcQuaid,

I originally started writing it at `homebrew-core` and while reading that issue template decided it very much did not fit there.  That issue template links to here:

> have a problem with brew install (or upgrade, reinstall) a single, official formula (not cask)? If it's a general brew problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new. If it's a brew cask problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.

- Link I followed (bypasses chooser, so no issue template): https://github.com/Homebrew/brew/issues/new
- Maybe that should update to this: https://github.com/Homebrew/brew/issues/new/choose

I have read [the templates for this repo](https://github.com/Homebrew/brew/tree/master/.github/ISSUE_TEMPLATE) now, I will need to reach out to the discourse as the most important step (how to reproduce it) is something I cannot figure out how to do.  It seems haphazard at best :/

**Comment by MikeMcQuaid** [comment]
Thanks! I'll update those links.
